internal/kms/keyprotect.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kms
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/ceph/ceph-csi/internal/util/k8s"
+
+	kp "github.com/IBM/keyprotect-go-client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	kmsTypeKeyProtectMetadata = "kp-metadata"
+
+	// KeyProtectMetadataDefaultSecretsName is the default name of the Kubernetes Secret
+	// that contains the credentials to access the Key Protect KMS. The name of
+	// the Secret can be configured by setting the `KMS_SECRET_NAME`
+	// option.
+	//
+	// #nosec:G101, value not credential, just references token.
+	KeyProtectMetadataDefaultSecretsName = "ceph-csi-kp-credentials"
+
+	// KeyProtectSecretNameKey contains the name of the Kubernetes Secret that has
+	// the credentials to access the Key ProtectKMS.
+	//
+	// #nosec:G101, no hardcoded secret, this is a configuration key.
+	KeyProtectSecretNameKey = "KMS_SECRET_NAME"
+	KeyProtectRegionKey     = "KP_REGION"
+
+	keyProtectServiceInstanceID = "KP_SERVICE_INSTANCE_ID"
+	// The following options are part of the Kubernetes Secrets.
+	// #nosec:G101, no hardcoded secrets, only configuration keys.
+	KeyProtectServiceAPIKey   = "KP_SERVICE_API_KEY"
+	KeyProtectCustomerRootKey = "KP_CUSTOMER_ROOT_KEY"
+
+	KeyProtectSessionToken = "KP_SESSION_TOKEN"
+	KeyProtectCRK          = "KP_CRK_ARN"
```

Comment:
this will be used only for KP, not for HPCS? can we have a generic name if this will be used for both?

Reply:
regardless we use hpcs/key protect, the API server is same which is `Key Protect`, so this is the generic name,